### PR TITLE
Function to return supported Kotlin version

### PR DESCRIPTION
**Pull request by kkocel**
Introduce the function that returns the supported Kotlin version as mentioned in: https://github.com/detekt/detekt/issues/6198#issuecomment-1710122711

### CODE REVIEW

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt [hunk 1]
```diff
@@ -139,3 +141,15 @@ class DetektPlugin : Plugin<Project> {
 const val CONFIGURATION_DETEKT = "detekt"
 const val CONFIGURATION_DETEKT_PLUGINS = "detektPlugins"
 const val USE_WORKER_API = "detekt.use.worker.api"
+
+fun getSupportedKotlinVersion(): String {
```
Comment (3flex) [inline detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt:145]: Can you please add an `@Incubating` annotation here?

**Review by 3flex — COMMENTED** [review]
I just want to note that while this will work in the majority of cases it is possible to change the version of detekt-cli that the plugin calls, and the resolved version of the CLI dependency might depend on a different Kotlin version which will cause failures again.

I think we can get the resolved Kotlin dependency version from Gradle though when I last tried this it didn't seem that easy. But in the meantime annotating with Incubating will at least flag in IDE that this isn't 100% stable or foolproof.

File: detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt (hunk 1 above)
Comment (kkocel) [reply detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt:147]: Done

**Comment by kkocel** [comment]
> I just want to note that while this will work in the majority of cases it is possible to change the version of detekt-cli that the plugin calls, and the resolved version of the CLI dependency might depend on a different Kotlin version which will cause failures again.

I thought about adding dependency to detekt-cli and query supported version from there but I assumed that Kotlin versions will be consistent across modules.

**Comment by kkocel** [comment]
@3flex is there anything here that I should change or it's good to be merged?

**Comment by BraisGabin** [comment]
I vote to merge this. They should not be unaligned. And if they do we could always change this implementation without any breaking change.

**Review by BraisGabin — APPROVED** [review]

**Comment by BraisGabin** [comment]
@kkocel could you also update the documentation pointing out how to use this workaround? And that it should only be needed if the user applies the spring dependency manager (I can't recall the exact name)

**Comment by kkocel** [comment]
@BraisGabin thank you. I added the relevant part in the _future_ docs and mentioned Spring Dependency Plugin there.

**Pull request merged**